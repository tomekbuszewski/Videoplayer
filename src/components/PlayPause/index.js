/**
 * @flow
 * @author tomekbuszewski
 * @since 2018-11-26
 */

import * as React from 'react';
import inject from 'react-jss';

import styles from './index.styles';

type Props = {
  classes: Object,
  onClick: Function,
  paused: boolean,
}

const PlayPause = ({ classes, onClick, paused }: Props) => (
  <button type="button" onClick={onClick}>{paused ? '►' : '❚❚'}</button>
);

export default inject(styles)(PlayPause);
