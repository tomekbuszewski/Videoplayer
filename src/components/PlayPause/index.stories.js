import * as React from 'react';

import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import PlayPause from './';

storiesOf('Atoms', module)
  .addDecorator(withKnobs)
  .add(
    'PlayPause',
    (() => <PlayPause />),
  );
