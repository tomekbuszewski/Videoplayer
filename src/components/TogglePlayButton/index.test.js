import * as React from 'react';
import renderer from 'react-test-renderer';

import PlayPause from '.';

describe('PlayPause tests', () => {
  it('should match snapshot while paused', () => {
    const component = renderer.create(<PlayPause paused onClick={() => false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot while not paused', () => {
    const component = renderer.create(<PlayPause paused={false} onClick={() => false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
