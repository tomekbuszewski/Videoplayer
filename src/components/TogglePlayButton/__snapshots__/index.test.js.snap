// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayPause tests should match snapshot while not paused 1`] = `
<button
  className="PlayPause-Button-0-1-1"
  onClick={[Function]}
  type="button"
/>
`;

exports[`PlayPause tests should match snapshot while paused 1`] = `
<button
  className="PlayPause-Button-0-1-1"
  onClick={[Function]}
  type="button"
/>
`;
